### PR TITLE
Combine selected crucible asset and selected wallet asset stores

### PR DESCRIPTION
**Pull request by alexgeorgousis**

### CODE REVIEW

File: src/store/transactions/useTransactions.tsx [hunk 1]
```diff
@@ -127,14 +124,15 @@ export const useTransactions = (): UseTransactions => {
         dispatchedAction.type === 'crucibles/withdrawFromCrucible/fulfilled'
       ) {
         batch(() => {
+          queryClient.invalidateQueries([
+            Queries.CRUCIBLE_ASSETS,
+            dispatchedAction.meta.arg.crucibleAddress
+          ]);
+          queryClient.invalidateQueries([Queries.WALLET_ASSETS]);
```
Comment (nuurcodes) [inline src/store/transactions/useTransactions.tsx:131]: I think we should add the chainId and account when invalidating, I've added the change in the crucibles refactor PR

File: src/store/selectedAssets.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    [dispatch]
+  );
+
+  const crucibleAssets = queryClient.getQueryData([
+    Queries.CRUCIBLE_ASSETS,
+    crucibleAddress
+  ]) as ERC20Token[];
+
+  const walletAssets = queryClient.getQueryData([
+    Queries.WALLET_ASSETS
```
Comment (nuurcodes) [inline src/store/selectedAssets.ts:63]: we would need chainId and account here

File: src/store/selectedAssets.ts [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    }
+  };
+
+  useEffect(() => {
+    updateSelectedAsset(
+      crucibleAssets,
+      selectedCrucibleAsset,
+      setSelectedCrucibleAsset
+    );
+  }, [crucibleAssets, selectedCrucibleAsset, setSelectedCrucibleAsset]);
```
Comment (nuurcodes) [inline src/store/selectedAssets.ts:92]: I tend to not include functions in useEffect deps (e.g. setSelectedCrucibleAsset, this also means you wont need useCallbacks above

File: src/store/crucibleAssets.ts [hunk 4]
```diff
@@ -1,198 +0,0 @@
-import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
```
Comment (nuurcodes) [inline src/store/crucibleAssets.ts:1]: RIP

File: src/hooks/query/useWalletAssets.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+import { useConfig } from 'store/config';
+import { ERC20Token } from 'types';
+import { Queries } from 'types/enum';
+
+export function useWalletAssets() {
+  const { account, chainId = 1, library } = useWeb3React();
+  const { etherscanApiKey } = useConfig(chainId).commonConfig;
+
+  return useQuery<ERC20Token[]>(
+    [Queries.WALLET_ASSETS],
```
Comment (nuurcodes) [inline src/hooks/query/useWalletAssets.ts:13]: chainId and account needed for the query, see crucibles refactor PR

File: src/components/crucible/detail/summary/index.tsx [hunk 6]
```diff
@@ -27,13 +30,32 @@ type Props = {
...
+  const { chainId = 1, library } = useWeb3React();
+  const { etherscanApiKey } = useConfig(chainId).commonConfig;
   const { cruciblesLoading } = useCrucibles();
   const { status: lockStatus, crucibleLocks } = useCrucibleLocks();
-  const { status: assetsStatus, crucibleAssets } = useCrucibleAssets();
-  const { chainId = 1 } = useWeb3React();
   const { rewardPrograms } = useRewardPrograms();
   const [isLargerThan768] = useMediaQuery('(min-width: 769px)');
 
+  const {
```
Comment (nuurcodes) [inline src/components/crucible/detail/summary/index.tsx:40]: can make use of the hook you created

**Review by nuurcodes — APPROVED** [review]
Awesome work 🔥  left a few comments

File: src/store/transactions/useTransactions.tsx (hunk 1 above)
Comment (alexgeorgousis) [reply src/store/transactions/useTransactions.tsx]: Done ✅

File: src/store/selectedAssets.ts (hunk 3 above)
Comment (alexgeorgousis) [reply src/store/selectedAssets.ts:97]: I like that idea - changed it 👍

File: src/store/crucibleAssets.ts (hunk 4 above)
Comment (alexgeorgousis) [reply src/store/crucibleAssets.ts:1]: ☮️

**Pull request merged**